### PR TITLE
support (de)serialization of PCA models

### PR DESCRIPTION
**Pull request by yitao-li**

### CODE REVIEW

File: tests/testthat/test-pca-serde.R [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        model <- cuda_ml_unserialize(model_state)
+
+        expect_equal(model$components, expected_components)
+        expect_equal(model$explained_variance, expected_expl_var)
+        expect_equal(
+          model$explained_variance_ratio, expected_expl_var_ratio
+        )
+        expect_equal(model$singular_values, expected_sg_vals)
+        expect_equal(model$mean, expected_m)
+        expect_equal(model$transformed_data, expected_tf_data)
```
Comment (t-kalinowski) [inline tests/testthat/test-pca-serde.R:34]: Do you anticipate this list of attributes accessible under `model$` to grow with future cuml releases? Will we get those additional attributes "for free"? If yes, would it make sense to just iterate through all elements of `as.list(model)` (or something equivalent?)

Flip side, are there any attributes you don't expect check to survive saving/restoring? If yes, maybe a `expect_null` or `expect_error`?

File: R/pca.R [hunk 2]
```diff
@@ -72,3 +72,17 @@ cuda_ml_pca <- function(x,
...
+cuda_ml_get_state.cuda_ml_pca <- function(model) {
+  model_state <- .pca_get_state(model)
+  class(model_state) <- c("cuda_ml_pca_model_state", class(model_state))
+
+  model_state
+}
+
+cuda_ml_set_state.cuda_ml_pca_model_state <- function(model_state) {
+  model <- .pca_set_state(model_state)
+  class(model) <- c("cuda_ml_pca", "cuda_ml_model", class(model))
```
Comment (t-kalinowski) [inline R/pca.R:85]: I think `class(model)` here returns `"list"`. Why do you want the model to inherit from `"list"`?

**Review by t-kalinowski — APPROVED** [review]
Added minor comments in line. LGTM!

File: R/pca.R (hunk 2 above)
Comment (yitao-li) [reply R/pca.R:85]: No reason in particular. It's just that at the moment there is no need to "hide" the fact that this model object is a list.

File: tests/testthat/test-pca-serde.R (hunk 1 above)
Comment (yitao-li) [reply tests/testthat/test-pca-serde.R:34]: I don't expect additional attributes here in future versions of cuML.

**Pull request merged**